app/lib/frontend/backend.dart

```diff
@@ -522,30 +522,35 @@ class GCloudPackageRepository extends PackageRepository {
       }
     });
 
-    final uploaderEmails =
-        await accountBackend.getEmailsOfUserIds(package.uploaders);
-
-    // Notify uploaders via e-mail that a new version has been published.
-    await emailSender.sendMessage(
-      createPackageUploadedEmail(
-        packageName: newVersion.package,
-        packageVersion: newVersion.version,
-        uploaderEmail: user.email,
-        authorizedUploaders:
-            uploaderEmails.map((email) => EmailAddress(null, email)).toList(),
-      ),
-    );
+    try {
+      final uploaderEmails =
+          await accountBackend.getEmailsOfUserIds(package.uploaders);
+
+      // Notify uploaders via e-mail that a new version has been published.
+      await emailSender.sendMessage(
+        createPackageUploadedEmail(
+          packageName: newVersion.package,
+          packageVersion: newVersion.version,
+          uploaderEmail: user.email,
+          authorizedUploaders:
+              uploaderEmails.map((email) => EmailAddress(null, email)).toList(),
+        ),
+      );
 
-    // Trigger analysis and dartdoc generation. Dependent packages can be left
-    // out here, because the dependency graph's background polling will pick up
-    // the new upload, and will trigger analysis for the dependent packages.
-    await analyzerClient
-        .triggerAnalysis(newVersion.package, newVersion.version, <String>{});
-    await dartdocClient
-        .triggerDartdoc(newVersion.package, newVersion.version, <String>{});
+      // Trigger analysis and dartdoc generation. Dependent packages can be left
+      // out here, because the dependency graph's background polling will pick up
+      // the new upload, and will trigger analysis for the dependent packages.
+      await analyzerClient
+          .triggerAnalysis(newVersion.package, newVersion.version, <String>{});
+      await dartdocClient
+          .triggerDartdoc(newVersion.package, newVersion.version, <String>{});
```

Comment:
Why not trigger these concurrently?

Reply:
We could even do emailing concurrently with this...

Reply:
updated to do that.